Annotation `javax.annotation.processing.Generated` is used when building with jdk11 despite `targetVersion=8`

we are building on jdk11 (but target a java 8 runtime) with the maven plugin and setting `<targetVersion>8</targetVersion>`

but getting the following compilation errors:
```
error: cannot find symbol
    symbol:   class Generated
    location: package javax.annotation.processing
```

so we have to add `jsonschema2pojo-jdk-annotation` to the classpath to workaround the problem.

however: when targetting java 8 it should not try to add the java9 annotation here imo:
https://github.com/joelittlejohn/jsonschema2pojo/blob/284c1e81556eaa0bd4515a186f25fc875384eef9/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/AnnotationHelper.java#L42-L46

i am guessing `Class.forName` works for the java9 classname because we are building on jdk11.

a more general solution that would also cover https://github.com/joelittlejohn/jsonschema2pojo/issues/1461 is to make the annotation class freely configurable with another mojo property like `generatedAnnotationClass`

it could default to `"auto"` and keep running the existing logic if desired

--- a/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/DefaultGenerationConfig.java
+++ b/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/DefaultGenerationConfig.java
@@ -25,6 +25,7 @@ import java.util.Iterator;
 import java.util.Map;
 
 import org.jsonschema2pojo.rules.RuleFactory;
+import org.jsonschema2pojo.util.JavaVersion;
 
 /**
  * A generation config that returns default values for all behavioural options.
@@ -348,11 +349,11 @@ public class DefaultGenerationConfig implements GenerationConfig {
     }
 
     /**
-     * @return <code>1.6</code>
+     * @return Current JVM version
      */
     @Override
     public String getTargetVersion() {
-        return "1.6";
+        return JavaVersion.parse(System.getProperty("java.version"));
     }
 
     /**

--- a/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/GenerationConfig.java
+++ b/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/GenerationConfig.java
@@ -618,7 +618,7 @@ public interface GenerationConfig {
 
   /**
    * Whether to mark generated classes with the annotation <code>javax.annotation.@Generated</code>
-   *
+   * (or <code>javax.annotation.processing.Generated</code> for Java 9 and later).
    */
   boolean isIncludeGeneratedAnnotation();
 

--- a/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/EnumRule.java
+++ b/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/EnumRule.java
@@ -151,7 +151,7 @@ public class EnumRule implements Rule<JClassContainer, JType> {
         EnumDefinition enumDefinition = buildEnumDefinition(nodeName, node, backingType);
 
         if(ruleFactory.getGenerationConfig() != null && ruleFactory.getGenerationConfig().isIncludeGeneratedAnnotation()) {
-            AnnotationHelper.addGeneratedAnnotation(_enum);
+            AnnotationHelper.addGeneratedAnnotation(ruleFactory.getGenerationConfig(), _enum);
         }
 
         JFieldVar valueField = addConstructorAndFields(enumDefinition, _enum);

--- a/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java
+++ b/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java
@@ -132,7 +132,7 @@ public class ObjectRule implements Rule<JPackage, JType> {
         }
        
         if (ruleFactory.getGenerationConfig().isIncludeGeneratedAnnotation()) {
-        	AnnotationHelper.addGeneratedAnnotation(jclass);
+            AnnotationHelper.addGeneratedAnnotation(ruleFactory.getGenerationConfig(), jclass);
         }
         if (ruleFactory.getGenerationConfig().isIncludeToString()) {
             addToString(jclass);

--- a/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/AnnotationHelper.java
+++ b/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/AnnotationHelper.java
@@ -16,6 +16,8 @@
 
 package org.jsonschema2pojo.util;
 
+import org.jsonschema2pojo.GenerationConfig;
+
 import com.sun.codemodel.JAnnotationUse;
 import com.sun.codemodel.JClass;
 import com.sun.codemodel.JDefinedClass;
@@ -39,8 +41,10 @@ public class AnnotationHelper {
 
     }
 
-    public static void addGeneratedAnnotation(JDefinedClass jclass) {
-        if (!tryToAnnotate(jclass, JAVA_9_GENERATED)) {
+    public static void addGeneratedAnnotation(GenerationConfig config, JDefinedClass jclass) {
+        if (JavaVersion.is9OrLater(config.getTargetVersion())) {
+            tryToAnnotate(jclass, JAVA_9_GENERATED);
+        } else {
             tryToAnnotate(jclass, JAVA_8_GENERATED);
         }
     }

--- a/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/JavaVersion.java
+++ b/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/JavaVersion.java
@@ -39,8 +39,12 @@ public class JavaVersion {
     }
 
     public static boolean is9OrLater(final String targetVersion) {
-        final Double v = Double.valueOf(targetVersion);
-        return (v >= 9) || (v < 2 && v >= 1.9);
+        if (isNotBlank(targetVersion)) {
+            final Double v = Double.valueOf(targetVersion);
+            return (v >= 9) || (v < 2 && v >= 1.9);
+        } else {
+            return false;
+        }
     }
 
 }

--- a/jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/JavaVersionTest.java
+++ b/jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/JavaVersionTest.java
@@ -33,6 +33,8 @@ public class JavaVersionTest {
 
     @Test
     public void testIs9OrLater() {
+        assertThat(JavaVersion.is9OrLater(null), is(false));
+        assertThat(JavaVersion.is9OrLater(""), is(false));
         assertThat(JavaVersion.is9OrLater("1.1"), is(false));
         assertThat(JavaVersion.is9OrLater("1.2"), is(false));
         assertThat(JavaVersion.is9OrLater("1.3"), is(false));

--- a/jsonschema2pojo-gradle-plugin/README.md
+++ b/jsonschema2pojo-gradle-plugin/README.md
@@ -76,6 +76,7 @@ jsonSchema2Pojo {
 
   // Whether to include a javax.annotation.Generated (Java 8 and lower) or
   // javax.annotation.processing.Generated (Java 9+) in on generated types (default true).
+  // See also: targetVersion.
   includeGeneratedAnnotation = true
 
   // Whether to generate builder-style methods of the form withXxx(value) (that return this),

--- a/jsonschema2pojo-integration-tests/pom.xml
+++ b/jsonschema2pojo-integration-tests/pom.xml
@@ -103,6 +103,12 @@
             <type>maven-plugin</type>
         </dependency>
         <dependency>
+            <groupId>org.jsonschema2pojo</groupId>
+            <artifactId>jsonschema2pojo-jdk-annotation</artifactId>
+            <version>0.0.1</version>
+            <type>jar</type>
+        </dependency>
+        <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-plugin-api</artifactId>
         </dependency>

--- a/jsonschema2pojo-integration-tests/src/test/java/org/jsonschema2pojo/integration/config/IncludeGeneratedAnnotationIT.java
+++ b/jsonschema2pojo-integration-tests/src/test/java/org/jsonschema2pojo/integration/config/IncludeGeneratedAnnotationIT.java
@@ -37,17 +37,25 @@ public class IncludeGeneratedAnnotationIT {
     public Jsonschema2PojoRule schemaRule = new Jsonschema2PojoRule();
 
     @Test
-    public void defaultConfig() throws ClassNotFoundException {
+    public void defaultConfigHasGeneratedAnnotationsOn() throws ClassNotFoundException {
         File source = schemaRule.generate(SCHEMA_PATH, TEST_PACKAGE);
+        assertThat(source, FileSearchMatcher.containsText("@Generated"));
 
-        assertThat(source, FileSearchMatcher.containsText("javax.annotation.Generated"));
+        File sourceJava8 = schemaRule.generate(SCHEMA_PATH, TEST_PACKAGE, config("targetVersion", "8"));
+        assertThat(sourceJava8, FileSearchMatcher.containsText("javax.annotation.Generated"));
+
+        File sourceJava9 = schemaRule.generate(SCHEMA_PATH, TEST_PACKAGE, config("targetVersion", "9"));
+        assertThat(sourceJava9, FileSearchMatcher.containsText("javax.annotation.processing.Generated"));
+
+        File sourceJava11 = schemaRule.generate(SCHEMA_PATH, TEST_PACKAGE, config("targetVersion", "11"));
+        assertThat(sourceJava11, FileSearchMatcher.containsText("javax.annotation.processing.Generated"));
     }
 
     @Test
     public void disabled() throws ClassNotFoundException {
         File source = schemaRule.generate(SCHEMA_PATH, TEST_PACKAGE, config(PROP_KEY, false));
 
-        assertThat(source, Matchers.not(FileSearchMatcher.containsText("javax.annotation.Generated")));
+        assertThat(source, Matchers.not(FileSearchMatcher.containsText("@Generated")));
     }
 
     @Test

--- a/jsonschema2pojo-maven-plugin/src/main/java/org/jsonschema2pojo/maven/Jsonschema2PojoMojo.java
+++ b/jsonschema2pojo-maven-plugin/src/main/java/org/jsonschema2pojo/maven/Jsonschema2PojoMojo.java
@@ -764,8 +764,7 @@ public class Jsonschema2PojoMojo extends AbstractMojo implements GenerationConfi
     /**
      * Whether to include a javax.annotation.Generated (Java 8 and
      * lower) or javax.annotation.processing.Generated (Java 9+) in
-     * on generated types.
-     *
+     * on generated types. See also: targetVersion.
      */
     @Parameter(property = "jsonschema2pojo.includeGeneratedAnnotation", defaultValue = "true")
     private boolean includeGeneratedAnnotation = true;
@@ -775,7 +774,6 @@ public class Jsonschema2PojoMojo extends AbstractMojo implements GenerationConfi
      * when adding <a href="http://jcp.org/en/jsr/detail?id=303">JSR-303</a> annotations to generated Java types.
      * This property works in collaboration with the {@link #isIncludeJsr303Annotations()} configuration option.
      * If the {@link #isIncludeJsr303Annotations()} returns {@code false}, then this configuration option will not affect anything.
-     *
      */
     @Parameter(property = "jsonschema2pojo.useJakartaValidation", defaultValue = "false")
     private boolean useJakartaValidation = false;

